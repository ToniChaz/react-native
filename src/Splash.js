/**
 * Weather App
 * Build with react-native
 * @search-weather
 */

import React, { Component } from 'react'
import { Animated, StyleSheet, View, Easing } from 'react-native'

class Splash extends Component {
	static navigationOptions = {
		header: null
	}

	constructor(props) {
		super(props)
		this.spinValue = new Animated.Value(0)
	}

	componentDidMount() {
		this.spin()
		this.props.navigation.navigate('InfoPage')
		setTimeout(() => {
			this.props.navigation.navigate('InfoPage')
		}, 2000)
	}

	spin() {
		this.spinValue.setValue(0)
		Animated.timing(
			this.spinValue,
			{
				toValue: 1,
				duration: 4000,
				easing: Easing.linear
			}
		).start(() => this.spin())
	}

	render() {
		const spin = this.spinValue.interpolate({
			inputRange: [0, 1],
			outputRange: ['0deg', '360deg']
		})
		return (
			<View style={styles.splashContainer}>
				<Animated.Image
					style={{
						width: 125,
						height: 110,
						transform: [{rotate: spin}]
					}}
					source={require('../images/logo.png')}
				/>
			</View>
		)
	}
}

const styles = StyleSheet.create({
	splashContainer: {
		flex: 1,
		backgroundColor: '#204',
		justifyContent: 'center',
		alignItems: 'center',
	}
})

export default Splash