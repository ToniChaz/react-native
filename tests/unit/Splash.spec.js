// Splash.spec.js

import 'react-native'
import React from 'react'
import Splash from '../../src/Splash.js'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

describe('Splash component', () => {
	test('renders correctly', () => {
		const tree = renderer.create(
			<Splash />
		)

		expect(tree.toJSON()).toMatchSnapshot()
	})
})