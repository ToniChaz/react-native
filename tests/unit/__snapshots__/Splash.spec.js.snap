// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#204",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 110,
        "transform": Array [
          Object {
            "rotate": "0deg",
          },
        ],
        "width": 125,
      }
    }
  />
</View>
`;
